### PR TITLE
CLB: Dynaheir, Inovker Adept and support (kills another non-keyword keyword)

### LINKED ISSUE #425: Dynaheir, Invoker Adept
https://twitter.com/Dunkatog/status/1527464276285263872
> Due to how they work, mana abilities can't be copied by anything. Dynaheir's printed text is missing an exception for mana abilities in the trigger condition, so it is receiving an update in the Oracle card reference to prevent them from causing it to trigger.

### PR DESCRIPTION
**Pull request by Northmoc**
closes #425

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/dynaheir_invoker_adept.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Dynaheir, Invoker Adept
+ManaCost:1 U R W
+Types:Legendary Creature Human Wizard
+PT:4/4
+K:Haste
+S:Mode$ ActivateAbilityAsIfHaste | Affected$ Creature.Other+YouCtrl | Description$ You may activate abilities of other creatures you control as though those creatures had haste.
+A:AB$ Effect | Cost$ T | Triggers$ ActivateTrig | Duration$ Permanent | SpellDescription$ When you next activate an ability that isn't a mana ability this turn by spending four or more mana to activate it, copy that ability. You may choose new targets for the copy.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/dynaheir_invoker_adept.txt:7]: No Duration$?

File: forge-gui/res/cardsfolder/t/thousand_year_elixir.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Name:Thousand-Year Elixir
 ManaCost:3
 Types:Artifact
-S:Mode$ Continuous | Affected$ Creature.YouCtrl | AddHiddenKeyword$ CARDNAME may activate abilities as though it has haste. | Description$ You may activate abilities of creatures you control as though those creatures had haste.
-A:AB$ Untap | Cost$ 1 T | ValidTgts$ Creature | TgtPrompt$ Select target creature. | SpellDescription$ Untap target creature.
+S:Mode$ ActivateAbilityAsIfHaste | Affected$ Creature.YouCtrl | Description$ You may activate abilities of creatures you control as though those creatures had haste.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/t/thousand_year_elixir.txt:4]: Creature you control should restrict to battlefield

File: forge-game/src/main/java/forge/game/staticability/StaticAbilityActivateAbilityAsIfHaste.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static boolean applyCanActivateAbility(final StaticAbility stAb, final Card card) {
+
+        if (!stAb.matchesValidParam("Affected", card)) {
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/staticability/StaticAbilityActivateAbilityAsIfHaste.java:48]: I would probably call it ValidCard

File: forge-gui/res/cardsfolder/upcoming/dynaheir_invoker_adept.txt (hunk 1 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/dynaheir_invoker_adept.txt:7]: I've got a `Duration$ Permanent` in there

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/dynaheir_invoker_adept.txt:7]: I meant why is it there?

The ability says "this turn"

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/dynaheir_invoker_adept.txt:7]: Also wrong TriggerDescription

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/dynaheir_invoker_adept.txt:7]: Gotcha! Thanks 😅

**Review by tool4ever — APPROVED** [review]

**Pull request merged**